scripts/scenario_01_waves.lua

```diff
@@ -191,9 +191,66 @@ function update(delta)
 	end
 	enemy_count = 0
 	friendly_count = 0
+	local order = nil
+	local order_target = nil
+	local ship_type = nil
 	for _, enemy in ipairs(enemyList) do
 		if enemy:isValid() then
 			enemy_count = enemy_count + 1
+			order = enemy:getOrder()
+			if order == "Defend Target" or order == "Attack" then
+				order_target = enemy:getOrderTarget()
+				if order_target ~= nil then
```

Comment:
I'm pretty sure the default AI behaviour drops from defend target or attack to something else if the target is gone. So this shouldn't be needed, or that code should be fixed instead.

Reply:
Removed target validation code. Testing seems to show it changes to "Idle" if the target is bad